instrumentation/jsp-2.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jsp/HttpJspPageInstrumentation.java

```diff
@@ -52,23 +52,20 @@ public static void onEnter(
         @Advice.Argument(0) HttpServletRequest req,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
-      context = tracer().startSpan(tracer().spanNameOnRender(req), SpanKind.INTERNAL);
-      tracer().onRender(context, req);
+
+      context = instrumenter().start(Java8BytecodeBridge.currentContext(), req);
```

Comment:
Can you add the `shouldStart()` check?

Reply:
Does it make sense here? When can it return `true` for INTERNAL spans?

Reply:
I think we should do that for "future compatibility" (and uniformity in all instrumentations) - if we ever introduce some more elaborate suppresion rules, or have a `shouldStart()` customization interface, etc. There's also the `setDisabled` method.